driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -519,12 +523,10 @@ public String toString() {
 
     public static class Factory {
 
-        private final ExecutorService bossExecutor = Executors.newCachedThreadPool();
-        private final ExecutorService workerExecutor = Executors.newCachedThreadPool();
         public final HashedWheelTimer timer = new HashedWheelTimer(new ThreadFactoryBuilder().setNameFormat("Timeouter-%d").build());
 
-        private final ChannelFactory channelFactory = new NioClientSocketChannelFactory(bossExecutor, workerExecutor);
-        private final ChannelGroup allChannels = new DefaultChannelGroup();
+        private final EventLoopGroup eventLoopGroup = new NioEventLoopGroup();
+        private final ChannelGroup allChannels = new DefaultChannelGroup(GlobalEventExecutor.INSTANCE);
```

Comment:
GlobalEventExecutor will create a single thread and stop it when there is no task pending for 1 second.  There is also a comment in it that says 

    Please note it is not scalable to schedule large number of tasks to this executor; use a dedicated executor.

In this case should we be not be sharing the EventExecutor per ChannelGroup instance or is it ok?